azure-migrate/dependencies-at-scale/AzMig_Dependencies.ps1

```diff
@@ -0,0 +1,676 @@
+        throw "Output CSV file path is not valid"
+    }
+
+    if (Test-path -Path $OutputCsvFile) {
+        throw "File $OutputCsvFile already exists. Specify a different name or path"
+    }
+	
+	if (-not ($OutputCsvFile -match ".*\.csv$")) {
+        throw "Output file specified is not CSV."    
+    }
+
+    #Fetching the Project Id
+    $projectId = Get-AzMigProject -ResourceGroupName $ResourceGroupName -ProjectName $ProjectName
+    if(-not $projectId) {
+        throw "Project ID is invalid"
+    }
+    
+    #Get Appliance Details
+    $applianceDetails = Get-AzMigAppliances -ResourceGroupName $ResourceGroupName -ProjectName $ProjectName
+    if(-not $applianceDetails) {
+        throw "Server Discovery Solution missing Appliance Details. Invalid Solution"
+    }
+
+    $appMap = @{}
+    foreach($row in $applianceDetails){    
+        $appMap[$row.properties_applianceName] = $row.id
+    }
+
+    $vmwareappliancemap = @{}
+    #Discard non-VMware appliances
+    #If Appliance name is passed get data only for that appliance
+    #If Appliance name is not passed , get data for all appliances in that project
+    if (-not $ApplianceName) {
+	    $appMap.GetEnumerator() | foreach {if($_.Value -match "VMwareSites|HyperVSites|ServerSites") {$vmwareappliancemap[$_.Key] = $_.Value}}
+    }
+    else { 
+	    $appMap.GetEnumerator() | foreach {if($_.Value -match "VMwareSites|HyperVSites|ServerSites" -and $_.Key -eq $ApplianceName) {$vmwareappliancemap[$_.Key] = $_.Value}}
+    }
+
+    Write-Debug -Message "Appliance count : $vmwareappliancemap.count"
+
+    if($vmwareappliancemap) {$vmwareappliancemap | Out-String | Write-Debug};
+    if (-not $vmwareappliancemap.count) {
+        throw "No VMware VMs discovered in project"
+    }
+    Write-Host "Please wait while the list of discovered machines is downloaded..."
+    
+    foreach ($item in $vmwareappliancemap.GetEnumerator()) {
+        $SiteId = $item.Value
+        $appliancename = $item.Key
+        Write-Debug -Message "Get machines for Site $SiteId"
+        $kqlResult =  Get-AzMigDiscoveredMachines -SiteId $SiteId -appliancename $appliancename -Filter $Filter
+
+        if ($kqlResult) {
+            $appliancename = $item.Key
+            Write-Host "Machines discovered for $appliancename"
+            $headers = $kqlResult[0].PSObject.Properties | Select-Object -ExpandProperty Name
+            $csvData = @()
+            foreach ($machine in $kqlResult) {
+                $row = [ordered]@{}       
+                foreach ($header in $headers) {
+                    $row[$header] = $($machine.$header)
+                }
+                $csvData += New-Object PSObject -Property $row
+            }
+            $csvData | Export-Csv -Path $OutputCsvFile -NoTypeInformation -Append
+            Write-Host "List of machines saved to" $OutputCsvFile
+        } 
+        else {
+           Write-Host "No machines discovered in the appliance $appliancename"
```

Comment:
Need to change this comment then , the comment should be something like "No machines discovered in the appliance $appliancename . Please check if you passed right input parameters and filters"

Reply:
I did'nt understand

Reply:
Write-Host "No machines discovered in the appliance $appliancename" 
let say if user passed a wrong filter , this print statement could be misleading since there could be machines in his appliance just that the filter passed can be incorrect.
Instead we should print something like 
"No machines discovered in the appliance $appliancename -  Please check if you passed right input parameters and filters"

Reply:
Okay